libs/ledgerjs/packages/hw-transport-node-hid-singleton/src/listenDevices.ts

```diff
@@ -1,82 +1,89 @@
-import usbDetect from "usb-detection";
+import { usb } from "usb";
 import { ledgerUSBVendorId } from "@ledgerhq/devices";
 import { log } from "@ledgerhq/logs";
+
 export type Device = {
   locationId: number;
   vendorId: number;
   productId: number;
   deviceName: string;
   manufacturer: string;
-  serialNumber: string;
+  serialNumber: number;
   deviceAddress: number;
 };
 
-const deviceToLog = ({ productId, locationId, deviceAddress }) =>
-  `productId=${productId} locationId=${locationId} deviceAddress=${deviceAddress}`;
+const deviceToLog = ({
+  deviceDescriptor: { idProduct },
+  busNumber,
+  deviceAddress,
+}) =>
+  `productId=${idProduct} busNumber=${busNumber} deviceAddress=${deviceAddress}`;
 
 let usbDebounce = 1000;
 export const setUsbDebounce = (n: number) => {
   usbDebounce = n;
 };
-let monitoring = false;
-
-const monitor = () => {
-  if (!monitoring) {
-    monitoring = true;
-    usbDetect.startMonitoring();
-  }
 
-  return () => {};
-};
-
-// No better way for now. see https://github.com/LedgerHQ/ledgerjs/issues/434
-process.on("exit", () => {
-  if (monitoring) {
-    // redeem the monitoring so the process can be terminated.
-    usbDetect.stopMonitoring();
-  }
+const mapRawDevice = ({
+  busNumber: locationId,
+  deviceAddress,
+  deviceDescriptor: {
+    idVendor: vendorId,
+    idProduct: productId,
+    iSerialNumber: serialNumber,
+  },
+}: usb.Device): Device => ({
+  locationId, // Nb we dont use this but the mapping might be incorrect.
+  vendorId,
+  productId,
+  deviceName: "",
```

Comment:
mmh, do you know if usb lib have some sort of deviceName still?
because with hid we were getting something like `Ledger Nano S` and it's not impossible that we use this information somewhere, 

i see it used at least on "cli" but hopefully it's just the CLI.

```
        name: e.device.deviceName || "",
```

Reply:
ultimately I wonder if we need to have the intermediary `Device` type, we could maybe return the "usb.Device" as is 🤔  to avoid need to set these empty fields

Reply:
I tried to have the minimum impact here, so as to not be a breaking change. If we modified the emitted signature maybe we are breaking other projects since it's an exported method. And I was unable to find the name at all on the legacy usb data.